### ISSUE TITLE
BBC: Blead Breaks experimentals

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#20862]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.43 running under perl 5.37.6.

-----------------------------------------------------------------

BBC: Blead Breaks experimentals

Please see http://fast-matrix.cpantesters.org/?dist=experimentals

---
**Flags**
- category=core
- severity=low
---
**Perl configuration**
```
Site configuration information for perl 5.37.10:

Configured by cpan at Sun Feb 26 21:41:46 EST 2023.

Summary of my perl5 (revision 5 version 37 subversion 10) configuration:
  Commit id: 10c7b2021d05bc5b69caa39f144a06954fa66d68
  Platform:
    osname=linux
    osvers=5.15.96-0-lts
    archname=x86_64-linux-thread-multi
    uname='linux cjg-alpine3 5.15.96-0-lts #1-alpine smp sat, 25 feb 2023 21:02:54 +0000 x86_64 linux '
    config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall -Duseithreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong'
    ccversion=''
    gccversion='12.2.1 20220924'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib /lib
    libs=-lpthread -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=/usr/lib/libc.a
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


---
@INC for perl 5.37.10:
    /home/cpan/bin/perl-blead/lib/site_perl/5.37.10/x86_64-linux-thread-multi
    /home/cpan/bin/perl-blead/lib/site_perl/5.37.10
    /home/cpan/bin/perl-blead/lib/5.37.10/x86_64-linux-thread-multi
    /home/cpan/bin/perl-blead/lib/5.37.10

---
Environment for perl 5.37.10:
    HOME=/home/cpan
    LANG=C.UTF-8
    LANGUAGE (unset)
    LC_COLLATE=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl-blead/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash
```

**Comment by demerphq** [comment]
@book - more smartmatch breakage

```
Configuring experimentals-0.019
Running Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for experimentals
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Test::More 0 ... Yes (1.302192)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.66)
Building and testing experimentals-0.019
cp lib/experimentals.pm blib/lib/experimentals.pm
PERL_DL_NONLAZY=1 "/home/yorton/perl5/perlbrew/perls/latest_blead/bin/perl5.37.10" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/no_experimentals.t ... ok

#   Failed test 'Got unexpected warning'
#   at t/use_experimentals.t line 3.
# given is deprecated at (eval 9) line 3.

#   Failed test 'Got unexpected warning'
#   at t/use_experimentals.t line 3.
# when is deprecated at (eval 9) line 4.

#   Failed test 'Got unexpected warning'
#   at t/use_experimentals.t line 3.
# Smartmatch is deprecated at (eval 11) line 3.
# Looks like you failed 3 tests of 11.
t/use_experimentals.t .. 
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/11 subtests 

Test Summary Report
-------------------
t/use_experimentals.t (Wstat: 768 (exited 3) Tests: 11 Failed: 3)
  Failed tests:  1-2, 5
  Non-zero exit status: 3
Files=2, Tests=13,  0 wallclock secs ( 0.03 usr  0.00 sys +  0.20 cusr  0.02 csys =  0.25 CPU)
Result: FAIL
Failed 1/2 test programs. 3/13 subtests failed.
make: *** [Makefile:846: test_dynamic] Error 3
-> FAIL Installing experimentals failed. See /home/yorton/.cpanm/work/1677491016.455583/build.log for details. Retry with --force to force install it.
```

**Comment by rjbs** [comment]
I filed https://rt.cpan.org/Ticket/Display.html?id=147648 .

**Comment by jkeenan** [comment]
> I filed https://rt.cpan.org/Ticket/Display.html?id=147648 .

[Upstream author indicates](https://rt.cpan.org/Ticket/Display.html?id=147648) he will address this problem once perl-5.38.0 is released.

**Comment by jkeenan** [comment]
Fixed with experimentals-0.020 released July 04 2023.
